### PR TITLE
Kinesis s3 sns.

### PR DESCRIPTION
**Pull request by thepug**
Needed: https://github.com/open-company/open-company-lib/pull/52

This change writes to a kinesis firehouse if an exception is thrown using SNS.  This is to help get around the limitations of SNS (size, 256kb limit, is the main concern here.)

Each service interested in the notification data from storage will have to use the new function provided in open-company-lib.

Service PRS:
https://github.com/open-company/open-company-notify/pull/7

https://github.com/open-company/open-company-change/pull/13

https://github.com/open-company/open-company-email/pull/40/

https://github.com/open-company/open-company-bot/pull/53

https://github.com/open-company/open-company-search/pull/13

Deploy with https://github.com/belucid/open-company-infrastructure/pull/96

To test:
- create a long post ~260k characters
- watch the aws cloudwatch logs or local logs for the `SNS failed with:`
- if the excpetion occurred, it would have been sent to kinesis
- [x] did the message show up in search?
- add a mention to the post
- [x] did it show up in email/bot as a notification?
- [x] Does the web client get a change message when the post is edited?

### CODE REVIEW

**Comment by thepug** [comment]
[SNS message too long for the storage notifications](https://trello.com/c/bNqg0J9x/1766-sns-message-too-long-for-the-storage-notifications)

File: src/oc/storage/async/notification.clj [hunk 1]
```diff
@@ -75,14 +76,23 @@
...
+  (let [subject (str (name (:notification-type trigger))
+                     " on " (name (:resource-type trigger))
+                     ": " (-> trigger :current :uuid))]
+    (try
+      (sns/publish
+       {:access-key config/aws-access-key-id
+        :secret-key config/aws-secret-access-key}
+       :topic-arn config/aws-sns-storage-topic-arn
+       :subject subject
+       :message (json/generate-string trigger {:pretty true}))
```
Comment (bago2k4) [inline src/oc/storage/async/notification.clj]: @thepug maybe `(json/generate-string trigger {:pretty true})` can go into a var in the let? it's called again if sns fails

File: src/oc/storage/util/search.clj [hunk 2]
```diff
@@ -37,12 +38,18 @@
...
+     {:access-key config/aws-access-key-id
+      :secret-key config/aws-secret-access-key}
+     config/aws-sqs-search-index-queue
+     (json/generate-string trigger {:pretty true}))
+    (catch Exception e
+      (timbre/info "SQS failed with: " e)
+      ;; If an exception occurred write to the kinesis firehouse.
+      (fh/put-record
+       config/aws-kinesis-reindex-stream-name
+       (json/generate-string trigger {:pretty true})))))
```
Comment (bago2k4) [inline src/oc/storage/util/search.clj]: @thepug same here

Comment (thepug) [reply src/oc/storage/util/search.clj]: updated

File: src/oc/storage/async/notification.clj (hunk 1 above)
Comment (thepug) [reply src/oc/storage/async/notification.clj]: updated

**Pull request merged**